uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/util/SslUtil.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.util;
+
+import com.aws.greengrass.testing.mqtt5.client.MqttLib;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+
+import java.io.BufferedInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.Security;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+public final class SslUtil {
+    private SslUtil() {
+    }
+
+    /**
+     * generate SSL socket factory.
+     *
+     * @param connectionParams MQTT connection parameters
+     * @throws IOException on errors
+     * @throws GeneralSecurityException on errors
+     */
+    public static SSLSocketFactory getSocketFactory(MqttLib.ConnectionParams connectionParams)
+            throws IOException, GeneralSecurityException {
+        return getSocketFactory(connectionParams.getCa(), connectionParams.getCert(), connectionParams.getKey());
+    }
+
+    /**
+     * generate SSL socket factory.
+     *
+     * @param caCrtFile client certification
+     * @param crtFile client certification
+     * @param keyFile client certification
+     * @throws IOException on errors
+     * @throws GeneralSecurityException on errors
+     */
+    @SuppressWarnings({"PMD.CloseResource", "PMD.AvoidFileStream"})
+    public static SSLSocketFactory getSocketFactory(final String caCrtFile, final String crtFile, final String keyFile)
+            throws IOException, GeneralSecurityException {
+        Security.addProvider(new BouncyCastleProvider());
+
+        // load CA certificate
+        X509Certificate caCert = null;
+
+        BufferedInputStream bis = new BufferedInputStream(Files.newInputStream(Paths.get(caCrtFile)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *newInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *newInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

Reply:
fixed